nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/clustering/FlowSynchronizationIT.java

```diff
@@ -424,30 +429,42 @@ public void testCanJoinClusterIfAllNodesMissingNar() throws NiFiClientException,
         assertTrue(getNifiClient().getProcessorClient().getProcessor(generate.getId()).getComponent().getExtensionMissing());
     }
 
+
     @Test
-    public void testCannotJoinIfMissingConnectionHasData() throws NiFiClientException, IOException, InterruptedException {
+    public void testCannotRemoveComponentsWhileNodeDisconnected() throws NiFiClientException, IOException, InterruptedException {
         final ProcessorEntity generate = getClientUtil().createProcessor("GenerateFlowFile");
         final ProcessorEntity terminate = getClientUtil().createProcessor("TerminateFlowFile");
         final ConnectionEntity connection = getClientUtil().createConnection(generate, terminate, "success");
 
-        getClientUtil().updateProcessorSchedulingPeriod(generate, "60 sec");
-
         // Shut down node 2
         disconnectNode(2);
+        waitForNodeState(2, NodeConnectionState.DISCONNECTED);
 
-        switchClientToNode(2);
-        getClientUtil().startProcessor(generate);
-        waitForQueueCount(connection.getId(), 1);
+        // Attempt to delete connection. It should throw an Exception.
+        try {
+            getNifiClient().getConnectionClient().deleteConnection(connection);
+            Assertions.fail("Was able to remove connection while node disconnected");
+        } catch (final Exception expected) {
+        }
```

Comment:
The `assertThrows` method is preferable to the try-catch approach that was necessary in earlier versions of JUnit.
```suggestion
        assertThrows(Exception.class, () -> getNifiClient().getConnectionClient().deleteConnection(connection));
```